### PR TITLE
Azure false failed order bug

### LINKED ISSUE #435: False Failed Order
This bug has happened in Azure Plugins when there is an Asynchronous operation. 
It happens because of the Asynchronous operation made by Azure SDK. If this asynchronous operation delays a lot, the Spawning process may ask the Plugin about the instance and it might not be ready. Therefore, the order fails due the fact that it seems that the instance has failed.

Image displaying the problem: [Image](https://lh3.googleusercontent.com/-uW3cW8Iv9iU/XrKyqx_kuBI/AAAAAAAAPZY/DOq9-gJYvusLedgNdeqliywQy2qkY3tLACK8BGAsYHg/s0/2020-05-06.png)
Image displaying the solution: [Image](https://lh3.googleusercontent.com/-8c5WPc1iEvA/XrK6wiFuZRI/AAAAAAAAPZk/GZyOEFCE8H0agLY52Y4Pd4aF1FzayAmMgCK8BGAsYHg/s0/2020-05-06.png)

### PR DESCRIPTION
**Pull request by fgan1**
## Description
Fixes #435 

## Proposed changes
- Build and store a list of Instances with "Creating" state.
- Create a class to cope with Azure asynchronous instance creation.

## Note
I've used a different approach and I would like you guys to take a look at it and evaluate it. Look at AzureAsync.

## Reminding
PRs order:
develop < this

### CODE REVIEW

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    /*
+    It must be used soon before the plugin makes asynchronous creation operation in the cloud.
+
+    @return It is a callback that it must be used when the cloud finish asynchronous creation operation in the cloud;
+    Suggestion: Use "finishCreationCallback" as attribute name.
+       */
+    // TODO(chico) - Add log debug
+    public Runnable startCreation(String instanceId) {
+        LOGGER.debug(Messages.Info.START_ASYNC_INSTANCE_CREATION_S);
```
Comment (jadsonluan) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java:31]: Format the message info string to fill the %s variable in the string.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        defineAsCreating(instanceId);
+        return () -> {
+            defineAsCreated(instanceId);
+            LOGGER.debug(Messages.Info.END_ASYNC_INSTANCE_CREATION_S);
```
Comment (jadsonluan) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java:35]: Format the message info string to fill the %s variable in the string.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java (hunk 1 above)
Comment (fgan1) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java]: Done

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java (hunk 2 above)
Comment (fgan1) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java]: Done

**Pull request merged**